ddtrace/appsec/_constants.py

```diff
@@ -41,85 +47,103 @@ def __getitem__(self, k: str) -> Any:
+    AUTO_LOGIN_EVENTS_SUCCESS_MODE: Literal[
+        "_dd.appsec.events.users.login.success.auto.mode"
+    ] = "_dd.appsec.events.users.login.success.auto.mode"
+    AUTO_LOGIN_EVENTS_FAILURE_MODE: Literal[
+        "_dd.appsec.events.users.login.failure.auto.mode"
+    ] = "_dd.appsec.events.users.login.failure.auto.mode"
+    BLOCKED: Literal["appsec.blocked"] = "appsec.blocked"
+    EVENT: Literal["appsec.event"] = "appsec.event"
+    AUTOMATIC_USER_EVENTS_TRACKING: Literal[
+        "DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING"
+    ] = "DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING"
+    AUTO_USER_INSTRUMENTATION_MODE: Literal[
+        "DD_APPSEC_AUTO_USER_INSTRUMENTATION_MODE"
+    ] = "DD_APPSEC_AUTO_USER_INSTRUMENTATION_MODE"
+    AUTO_USER_INSTRUMENTATION_MODE_ENABLED: Literal[
+        "DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING_ENABLED"
+    ] = "DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING_ENABLED"
+    USER_MODEL_LOGIN_FIELD: Literal["DD_USER_MODEL_LOGIN_FIELD"] = "DD_USER_MODEL_LOGIN_FIELD"
+    USER_MODEL_EMAIL_FIELD: Literal["DD_USER_MODEL_EMAIL_FIELD"] = "DD_USER_MODEL_EMAIL_FIELD"
+    USER_MODEL_NAME_FIELD: Literal["DD_USER_MODEL_NAME_FIELD"] = "DD_USER_MODEL_NAME_FIELD"
+    PROPAGATION_HEADER: Literal["_dd.p.appsec"] = "_dd.p.appsec"
+    OBFUSCATION_PARAMETER_KEY_REGEXP: Literal[
+        "DD_APPSEC_OBFUSCATION_PARAMETER_KEY_REGEXP"
+    ] = "DD_APPSEC_OBFUSCATION_PARAMETER_KEY_REGEXP"
+    OBFUSCATION_PARAMETER_VALUE_REGEXP: Literal[
+        "DD_APPSEC_OBFUSCATION_PARAMETER_VALUE_REGEXP"
+    ] = "DD_APPSEC_OBFUSCATION_PARAMETER_VALUE_REGEXP"
 
 
 class IAST(metaclass=Constant_Class):
     """Specific constants for IAST"""
 
-    ENV = "DD_IAST_ENABLED"
-    ENV_DEBUG = "_DD_IAST_DEBUG"
-    TELEMETRY_REPORT_LVL = "DD_IAST_TELEMETRY_VERBOSITY"
-    LAZY_TAINT = "_DD_IAST_LAZY_TAINT"
-    JSON = "_dd.iast.json"
-    ENABLED = "_dd.iast.enabled"
-    CONTEXT_KEY = "_iast_data"
-    PATCH_MODULES = "_DD_IAST_PATCH_MODULES"
-    DENY_MODULES = "_DD_IAST_DENY_MODULES"
-    SEP_MODULES = ","
-    REQUEST_IAST_ENABLED = "_dd.iast.request_enabled"
-    TEXT_TYPES = (str, bytes, bytearray)
+    ENV: Literal["DD_IAST_ENABLED"] = "DD_IAST_ENABLED"
+    ENV_DEBUG: Literal["_DD_IAST_DEBUG"] = "_DD_IAST_DEBUG"
+    TELEMETRY_REPORT_LVL: Literal["DD_IAST_TELEMETRY_VERBOSITY"] = "DD_IAST_TELEMETRY_VERBOSITY"
+    LAZY_TAINT: Literal["_DD_IAST_LAZY_TAINT"] = "_DD_IAST_LAZY_TAINT"
+    JSON: Literal["_dd.iast.json"] = "_dd.iast.json"
+    ENABLED: Literal["_dd.iast.enabled"] = "_dd.iast.enabled"
+    CONTEXT_KEY: Literal["_iast_data"] = "_iast_data"
+    PATCH_MODULES: Literal["_DD_IAST_PATCH_MODULES"] = "_DD_IAST_PATCH_MODULES"
+    DENY_MODULES: Literal["_DD_IAST_DENY_MODULES"] = "_DD_IAST_DENY_MODULES"
+    SEP_MODULES: Literal[","] = ","
+    REQUEST_IAST_ENABLED: Literal["_dd.iast.request_enabled"] = "_dd.iast.request_enabled"
+    TEXT_TYPES: Literal["str", "bytes", "bytearray"] = (str, bytes, bytearray)
```

Comment:
This type needs to be changed to a Literal tuple that contains those those string values.

Reply:
See: https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/66375/workflows/30c58678-7352-4dcf-a1db-1b2efbfb6fc0/jobs/4079892